stable/external-dns/templates/deployment.yaml

```diff
@@ -195,6 +195,19 @@ spec:
         - name: CF_API_EMAIL
           value: {{ .Values.cloudflare.email | quote }}
         {{- end }}
+        # CoreDNS environment variables
```

Comment:
What do you think about refactoring it in the following way?
- First of all, use the `.Values.provider "coredns"` condition for everything, like a first level, in this way we can avoid the `and` and we only check this condition one time:
```yaml
        # CoreDNS environment variables
        {{- if eq .Values.provider "coredns" }}
        - name: ETCD_URLS
          value: {{ .Values.coredns.etcdEndpoints | quote }}
        {{- if .Values.coredns.etcdTLS.enabled }}
        - name: ETCD_CERT_FILE
          value: {{ .Values.coredns.etcdTLS.mountPath }}/{{ .Values.coredns.etcdTLS.certFilename }}
        - name: ETCD_KEY_FILE
          value: {{ .Values.coredns.etcdTLS.mountPath }}/{{ .Values.coredns.etcdTLS.keyFilename }}
        - name: ETCD_CA_FILE
          value: {{ .Values.coredns.etcdTLS.mountPath }}/{{ .Values.coredns.etcdTLS.caFilename }}
        {{- end }}
        {{- end }}
```
- Second, we should ensure each value is present, as `mountPath` is repeated, we can do it at the same level than the `enabled`:
```yaml
        # CoreDNS environment variables
        {{- if eq .Values.provider "coredns" }}
        - name: ETCD_URLS
          value: {{ .Values.coredns.etcdEndpoints | quote }}
        {{- if and (.Values.coredns.etcdTLS.enabled) (.Values.coredns.etcdTLS.mountPath) }}
        {{- if .Values.coredns.etcdTLS.certFilename }}
        - name: ETCD_CERT_FILE
          value: {{ .Values.coredns.etcdTLS.mountPath }}/{{ .Values.coredns.etcdTLS.certFilename }}
        {{- end }}
        {{- if .Values.coredns.etcdTLS.keyFilename }}
        - name: ETCD_KEY_FILE
          value: {{ .Values.coredns.etcdTLS.mountPath }}/{{ .Values.coredns.etcdTLS.keyFilename }}
        {{- end }}
        {{- if .Values.coredns.etcdTLS.caFilename }}
        - name: ETCD_CA_FILE
          value: {{ .Values.coredns.etcdTLS.mountPath }}/{{ .Values.coredns.etcdTLS.caFilename }}
        {{- end }}
        {{- end }}
        {{- end }}
```
Take into account that I didn't test those changes, please use it as pseudo-code

Reply:
> First of all, use the .Values.provider "coredns" condition for everything..

agree, that makes a perfect sense

> Second, we should ensure each value is present...

it's not a problem to add what you are asking for but please consider my comments first:
* I expect operator to understand that if etcdTLS.enabled is true valid certificate data values should be present (Secret, mountPaths etc.)
* to be on a safe side defaults for all of those values are present in the values.yaml
* we anyway do not have nice and easy way to check if files with provided names are present in the Secret
* with that logic we need to add "if's" for all the values from values.yaml which for my opinion will look kinda awkward :)

@carrodher what do you think concerning second point?

Reply:
Ok, makes sense, as it is set by default in both values we can suppose that this value is not empty